### DISCUSSION TITLE
AssumeRoleAsync is not supported by the Language

### DISCUSSION THREAD

**Discussion by AXDevTimothy** [abuzuhri/Amazon-SP-API-CSharp#208] [Q&A]
I am forced to use visual studio 2013 with C# 5.0.  I am using .Net framework 4.5.2 as it seems to be the most compatible with the dependencies I need.  I am getting an error in the project saying that the AssumeRoleAsync method is not supported by the language.  Any ideas?

![image](https://user-images.githubusercontent.com/106777343/172677049-71812200-cfee-4314-8b99-fcf696e10654.png)

**Comment by abuzuhri** [comment]
Can you try to use nuget only and check if that will work with you

**Reply by AXDevTimothy** [reply]
I must use visual studio 2013 which is incompatible with your nuget solution.  I pulled your project into a 2022 visual studio environment and then copied the class over manually to my 2013 environment. I needed to also build out logic for the vendor retail procurement and vendor direct fulfillment which were not in your current version.  Everything compiles OK with the exception of that one line of code.  Do you know what is in that AssumeRoleAsync method? Maybe i can recreate it in VS 2013.

**Reply by abuzuhri** [reply]
Look to dependency AWSSDK.SecurityToken it’s came from this nuget
